async-nats/src/lib.rs

```diff
@@ -769,21 +780,48 @@ pub struct Message {
 pub struct Subscriber {
     _sid: u64,
     receiver: mpsc::Receiver<Message>,
+    sender: mpsc::Sender<ClientOp>,
 }
 
 impl Subscriber {
-    fn new(sid: u64, receiver: mpsc::Receiver<Message>) -> Subscriber {
+    fn new(
+        sid: u64,
+        sender: mpsc::Sender<ClientOp>,
+        receiver: mpsc::Receiver<Message>,
+    ) -> Subscriber {
         Subscriber {
             _sid: sid,
+            sender,
             receiver,
         }
     }
+
+    /// Unsubscribes from subscription, draining all remaining messages.
+    ///
+    /// # Examples
+    /// ```
+    /// # #[tokio::main]
+    /// # async fn unsubscribe() -> Result<(), Box<dyn std::error::Error>> {
+    /// let mut client = async_nats::connect("demo.nats.io").await?;
+    ///
+    /// let mut subscriber = client.subscribe("foo".into()).await?;
+    ///
+    ///  subscriber.unsubscribe();
+    /// # Ok(())
+    /// # }
+    pub fn unsubscribe(self) {
+        drop(self)
+    }
 }
 
 impl Drop for Subscriber {
     fn drop(&mut self) {
         // Can we get away with just closing, and then handling that on the sender side?
         self.receiver.close();
+        let handle = Handle::current();
```

Comment:
If we're not getting anything out of this, might as-well just let this run async in a task?